pkg/controller/azvolume.go

```diff
@@ -272,43 +272,39 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *azdiskv
 		var attachments []azdiskv1beta2.AzVolumeAttachment
 		attachments, err = r.cleanUpAzVolumeAttachmentByVolume(deleteCtx, azVolume.Name, azvolume, azureutils.AllRoles, mode)
 		if err == nil {
-			var wg sync.WaitGroup
-			errors := make([]error, len(attachments))
-			numErrors := uint32(0)
+			errorMessageCh := make(chan string)
```

Comment:
Create a buffered channel with a buffer size equal to the number of attachments, i.e. `make(chan string, len(attachments))`.

Reply:
Oh, I see the benefit of buffered channel here! I will first consider buffered next time. Done.